plugins/woocommerce/src/Admin/Features/MarketingRecommendations/Init.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Automattic\WooCommerce\Admin\Features\MarketingRecommendations;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Marketing Recommendations engine.
+ * This goes through the specs and gets marketing recommendations.
+ */
+class Init {
+	/**
+	 * Slug of the category specifying marketing extensions on the Woo.com store.
+	 *
+	 * @var string
+	 */
+	const MARKETING_EXTENSION_CATEGORY_SLUG = 'marketing';
+
+	/**
+	 * Slug of the subcategory specifying marketing channels on the Woo.com store.
+	 *
+	 * @var string
+	 */
+	const MARKETING_CHANNEL_SUBCATEGORY_SLUG = 'sales-channels';
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_action( 'woocommerce_updated', array( __CLASS__, 'delete_specs_transient' ) );
+	}
+
+	/**
+	 * Delete the specs transient.
+	 */
+	public static function delete_specs_transient() {
+		MarketingRecommendationsDataSourcePoller::get_instance()->delete_specs_transient();
+	}
+
+	/**
+	 * Get specs or fetch remotely if they don't exist.
+	 */
+	public static function get_specs() {
+		if ( 'no' === get_option( 'woocommerce_show_marketplace_suggestions', 'yes' ) ) {
+			return DefaultMarketingRecommendations::get_all();
+		}
+		$specs = MarketingRecommendationsDataSourcePoller::get_instance()->get_specs_from_data_sources();
+
+		// Fetch specs if they don't yet exist.
+		if ( false === $specs || ! is_array( $specs ) || 0 === count( $specs ) ) {
```

Comment:
```suggestion
		if ( ! is_array( $specs ) || 0 === count( $specs ) ) {
```
I think that you don't need to check directly for `false` when you are checking for an array with `is_array`. `is_array` will return `false` when `$specs` will be `false` thus making this equivalent.

Reply:
Good point! I also tested this and it works. 👍